### ISSUE TITLE
Bare-bones implementation of "Request Changes"

### ISSUE THREAD

**Issue by andrea-williams** [bcgov/cas-registration#375]
#### Description:

We want a simplified version of the "Request Changes" feature as part of MVP. 

Details of what changes are requested will be emailed to the external user by CAS staff (outside of the MVP Registration app). This ticket is only about setting the status of a database record to "Changes Requested".

#### Acceptance Criteria:

Given I am a CAS internal user who is logged in to the Registration app
When I click the "Request Changes" button for a Pending Operator or Operation
Then I am asked to confirm my request for changes. If confirmed, the status of that Operator or Operation is set to "Changes Requested" in the database. If not confirmed, the status of the Operator/Operation remains as it was before.

Given I am an external user who is logged in to the Registration app and is either an "admin" or a "reporter" for my Operator
When I view the list of Operations associated with my Operator
Then any Operations with a status of "Changes Requested" will be visible in the table (as shown in https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=1341%3A4450&mode=design&t=b0MX1lfLnGp4Rch3-1)

Given I am an external user who is logged in to the Registration app and is either an "admin" or "reporter" for my Operator
When an Operation associated with my Operator has a status of "Changes Requested"
Then the Operation form is editable so I can re-submit the form for CAS re-review.

#### Development Checklist:

- [ ] All wireframes relevant to Request Changes are here: https://www.figma.com/file/LsDmLDCdnJqI8UahyqvOD8/OBPS-App-Designs?type=design&node-id=3055%3A577&mode=design&t=b0MX1lfLnGp4Rch3-1
- [x] add enums for "Draft", "Pending", "Approved", "Rejected", and "Changes Requested" for both the Operator and Operation models
- [x] UI includes workflow to confirm user's choice to Request Changes
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by Dianadec** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Team-Moose-Sprint-Board-60d216175a2689000ec774d0/issues/bcgov/cas-registration/375?planning-poker) with Zenhub @andrea-williams @BCerki @dleard @JoshLarouche @marcellmueller @Sepehr-Sobhani @shon-button

**Comment by andrea-williams** [comment]
Review (re: "PO Review" column) - Request Changes feature works as desired in the context of Operations. It's also been implemented for new Operator records - waiting on decision from @hannavovk regarding whether internal users need to be able to request changes on an Operator record (will have knock-on impacts for existing UI)
